python/tests/avro/test_encoder.py

```diff
@@ -0,0 +1,230 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import datetime
+import io
+import struct
+from decimal import Decimal
+
+from pyiceberg.avro.encoder import BinaryEncoder
+
+
+def zigzag_encode(datum: int) -> bytes:
+    result = []
+    datum = (datum << 1) ^ (datum >> 63)
+    while (datum & ~0x7F) != 0:
+        result.append(struct.pack("B", (datum & 0x7F) | 0x80))
+        datum >>= 7
+    result.append(struct.pack("B", datum))
+    return b"".join(result)
+
+
+def test_write() -> None:
+    output = io.BytesIO()
+    encoder = BinaryEncoder(output)
+
+    _input = b"\x12\x34\x56"
+
+    encoder.write(_input)
+
+    assert output.getbuffer() == _input
+
+
+def test_write_boolean() -> None:
+    output = io.BytesIO()
+    encoder = BinaryEncoder(output)
+
+    encoder.write_boolean(True)
+    encoder.write_boolean(False)
+
+    assert output.getbuffer() == struct.pack("??", True, False)
+
+
+def test_write_int() -> None:
+    output = io.BytesIO()
+    encoder = BinaryEncoder(output)
+
+    _1byte_input = 2
+    _2byte_input = 7466
+    _3byte_input = 523490
+    _4byte_input = 86561570
+    _5byte_input = 2510416930
+    _6byte_input = 734929016866
+    _7byte_input = 135081528772642
+    _8byte_input = 35124861473277986
+
+    encoder.write_int(_1byte_input)
+    encoder.write_int(_2byte_input)
+    encoder.write_int(_3byte_input)
+    encoder.write_int(_4byte_input)
+    encoder.write_int(_5byte_input)
+    encoder.write_int(_6byte_input)
+    encoder.write_int(_7byte_input)
+    encoder.write_int(_8byte_input)
+
+    buffer = output.getbuffer()
+
+    assert buffer[0:1] == zigzag_encode(_1byte_input)
```

Comment:
Instead of this `zigzag_encode`. I think it makes more sense to assert against the actual bytes. The `zigzag_encode` is very similar to the actual encoding code, therefore we don't know if it is correct.